src/code.c

```diff
@@ -3233,9 +3168,6 @@ static Int InitKernel (
     MakeBagTypePublic(T_BODY);
 #endif
 
-    /* make the result variable known to Gasman                            */
-    InitGlobalBag(&CS(CodeResult), "CodeResult");
```

Comment:
@ChrisJefferson see, `CodeResult` is not handled by `InitGlobalBag` anymore.

Any other concerns? I am happy to answer. Of course this PR is rather too big, though I think reviewing it commit by commit is just barely manageable (the bigger commits are mostly a result of search and replace)